### PR TITLE
thermomachine upgrades

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
You can now upgrade the thermomachines by using 3 sheets of metallic h2 on them.
This upgrade allows them to consume all the heat without needing to connect the waste port to any gas source.

## Why It's Good For The Game
Upgradeable thermomachines and more uses for the metallic hydrogen

## Changelog

:cl:
expansion: thanks to the latest discoveries, metallic hydrogen can be installed inside the thermomachines to allow their use without needing a thermal port connected to a gas reservoir (a total of 3 metallic h2 sheets are required, obtain them from the local crystallizer, warranty void if applied to other machines)
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
so is this like old freezer behaviour? if so, that's kinda based

**Comment by Colovorat** [comment]
actual good ghilker pr wtf??????

File: code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm [hunk 1]
```diff
@@ -126,6 +144,12 @@
 	. += span_notice("-use a wrench with left-click to rotate [src] and right-click to unanchor it.")
 	. += span_notice("-use a multitool with left-click to change the piping layer and right-click to change the piping color.")
 	. += span_notice("The thermostat is set to [target_temperature]K ([(T0C-target_temperature)*-1]C).")
+
+	if(!has_metalh2)
+		. += span_notice("Can be upgraded with 2 sheets of metallic hydrogen.")
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm:149]: 2? or 3

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
I'm ok with this in theory. Not really sure there's a purpose to thermomachines having to manage cooling heat and all, at least with the restrictions I've set, but there's nothing inherently wrong with this, and that conversation is really tiring

File: code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm:149]: whoops is 3, i'll fix it

Comment (Ghilker) [inline code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm:149]:
```suggestion
		. += span_notice("Can be upgraded with 3 sheets of metallic hydrogen.")
```

**Comment by LemonInTheDark** [comment]
Will likely walk this back totally in future, but it's good for now

**Pull request merged**